llvm/lib/SandboxIR/SandboxIR.cpp

```diff
@@ -747,6 +747,38 @@ void StoreInst::dump() const {
 }
 #endif // NDEBUG
 
+UnreachableInst *UnreachableInst::create(Instruction *InsertBefore,
+                                         Context &Ctx) {
+  auto &Builder = Ctx.getLLVMIRBuilder();
+  Builder.SetInsertPoint(cast<llvm::Instruction>(InsertBefore->Val));
```

Comment:
Ah we also need to change this with:
```
  lvm::Instruction *LLVMBefore = InsertBefore->getTopmostLLVMInstruction(); 
 Builder.SetInsertPoint(LLVMBefore);
```
It shouldn't change the behavior for now, but once we introduce Multi-Instructions this won't work correctly.

I noticed that this is wrong in a few places, I will push a fix at some point.

Reply:
and this change is only for the create functions with the InsertBefore param?

Reply:
added the change, thanks for the suggestion!

Reply:
So from now on, should I always do

```
  auto &Builder = Ctx.getLLVMIRBuilder();
  llvm::Instruction *LLVMBefore = InsertBefore->getTopmostLLVMInstruction();
  Builder.SetInsertPoint(LLVMBefore);
```
for future implementations?

Reply:
> and this change is only for the create functions with the InsertBefore param?

Yes. For some context: If `InsertBefore` is a sandboxir Instruction that contains more than one LLVM Instruction, then we should always get its topmost LLVM Instruction and insert before it.

> if it's okay with you, I can create a NFC pr for those changes.

Sure, go for it.

> So from now on, should I always do
> ```
>  auto &Builder = Ctx.getLLVMIRBuilder();
>  llvm::Instruction *LLVMBefore = InsertBefore->getTopmostLLVMInstruction();
>  Builder.SetInsertPoint(LLVMBefore);
>```
> for future implementations?

Yes, this pattern should exist already in a couple of places.

Reply:
Understood.